src/ovi/Provider.php

```diff
@@ -0,0 +1,468 @@
+
+                        case 'email':
+                        $var = substr($var, 0, 254);
+                        $filter = FILTER_SANITIZE_EMAIL;
+                        break;
+
+                        case 'string':
+                        $filter = FILTER_SANITIZE_STRING;
+                        break;
+
+                        case 'boolean':
+                        $filter = FILTER_VALIDATE_BOOLEAN;
+                        break;
+
+                        default:
+                        $filter = FILTER_SANITIZE_STRING;
+                        $flags = FILTER_FLAG_NO_ENCODE_QUOTES;
+                        break;
+                }
+
+        $output = filter_var($var, $filter, $flags);
+        if($type != 'pass'){
+                $output = str_replace(array('\\', "\0", "\n", "\r", "'", '"', "\x1a","=","`"), array('', '', '', '', '', '', '','','','',''), $output);
+        }else{
+                $output = str_replace(array('\\', "\0", "\n", "\r", "\x1a","`"), array('', '', '', '', '', '', '','','','',''), $output);
+        }
+                return($output);
+        }
+
+
+    public function ovipanel_CurlExecute($input_tag, $request)
+   {
+    $ip = runtime_sanatizeItem::sanatizeItem(trim($this->configuration->ip),"IP");
+    $api_key = runtime_sanatizeItem:: sanatizeItem(trim($this->configuration->api_key),"string");
+    $serverusername = runtime_sanatizeItem::sanatizeItem(trim($this->configuration->adminusername),"string");
+    $serverpassword = runtime_sanatizeItem::sanatizeItem(trim($this->configuration->adminpassword),"pass");
+
+    $input_xml='<?xml version="1.0" encoding="UTF-8" ?>'.
+           '<xmws>'.
+           '<apikey>'.$api_key.'</apikey>'.
+           '<request>'.$request.'</request>'.
+           '<authuser>'.$serverusername.'</authuser>'.
+           '<authpass>'.$serverpassword.'</authpass>'.
+           '<content>' . $input_tag . '</content>'.
+           '</xmws>';
+
+    $url="http://$ip:2086/bin/oviapi.php?module=manage_clients";
+ //   ovipanel_MainErrorLog("Request URL : $url");
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_POSTFIELDS, "xmlRequest=" . $input_xml);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 60);
+    curl_setopt($ch, CURLOPT_TIMEOUT, 60);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+    $data = curl_exec($ch);
+    curl_close($ch);
+
+  //  ovipanel_MainErrorLog("Request $request");
+  //  ovipanel_MainErrorLog("input_xml : $input_xml");
+ //   ovipanel_MainErrorLog("Resposne $data");
+
+    return $data;
+
+
+}
+    public function ovipanel_UsernameExists($username)
+  {
+    $input_tag='<username>'.$username.'</username>';
+    $data = $this->ovipanel_CurlExecute($input_tag, "UsernameExists");
+    $data = json_decode($data, true);
+    return isset($data['success']) ? $data['success'] : false;
```

Comment:
It seems boolean `false` and string `"false"` are being used interchangeably which often leads to bugs. In this method i'd recommend strictly returning boolean to avoid any possibility of this issue.